### PR TITLE
Add SLJIT_ATOMIC_USE_LS/CAS support to LoongArch

### PR DESCRIPTION
**Pull request by lrzlin**
This fix test failure on LoongArch platform that support AMCAS. Tested under QEMU & Loongson 3A6000.

BTW, I do think adding a CI that tells us whether a PR will break some architecture or not is a need, which reduces our working strength, and also let us fix that failure in time.

### CODE REVIEW

File: sljit_src/sljitNativeLOONGARCH_64.c [hunk 1]
```diff
@@ -3625,7 +3627,10 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_load(struct sljit_compiler
 	CHECK_ERROR();
 	CHECK(check_sljit_emit_atomic_load(compiler, op, dst_reg, mem_reg));
 
-	if (!(LOONGARCH_CFG2_LAMCAS & get_cpu_features(GET_CFG2))) {
+	if (op & SLJIT_ATOMIC_USE_CAS && !LOONGARCH_SUPPORT_AMCAS)
```
Comment (zherczeg) [inline sljit_src/sljitNativeLOONGARCH_64.c:3630]: I prefer `()` around & to make it clearer. Similar cases below.

**Review by zherczeg — CHANGES_REQUESTED** [review]

File: sljit_src/sljitNativeLOONGARCH_64.c (hunk 1 above)
Comment (lrzlin) [reply sljit_src/sljitNativeLOONGARCH_64.c]: Added.

File: sljit_src/sljitNativeLOONGARCH_64.c [hunk 2]
```diff
@@ -3690,7 +3695,10 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_store(struct sljit_compiler
 	CHECK_ERROR();
 	CHECK(check_sljit_emit_atomic_store(compiler, op, src_reg, mem_reg, temp_reg));
 
-	if (!(LOONGARCH_CFG2_LAMCAS & get_cpu_features(GET_CFG2))) {
+	if ((op & SLJIT_ATOMIC_USE_CAS) && !LOONGARCH_SUPPORT_AMCAS)
```
Comment (zherczeg) [inline sljit_src/sljitNativeLOONGARCH_64.c:3698]: `LOONGARCH_SUPPORT_AMCAS` evaluated twice. `if (op & SLJIT_ATOMIC_USE_CAS) ..` would be moved into the if below, then one evaluation is enough.

Comment (lrzlin) [reply sljit_src/sljitNativeLOONGARCH_64.c]: Fixed, thanks.

**Review by zherczeg — APPROVED** [review]
LGTM

**Comment by zherczeg** [comment]
Thank you!

**Pull request merged**